File: edge-cases/java/anonymous-class/Sample.java
package edge.anonymous;

import java.util.Comparator;

public final class Sample {
    public Comparator<String> bySuffix(String suffix) {
        return new Comparator<String>() {
            @Override
            public int compare(String left, String right) {
                return (left + suffix).compareTo(right + suffix);
            }
        };
    }
}

File: fixtures/edge-static-j2k/anonymous-class/Sample.kt
package edge.anonymous

import java.util.Comparator

class Sample {
    fun bySuffix(suffix: String): Comparator<String> {
        return object : Comparator<String> {
            override fun compare(left: String, right: String): Int {
                return (left + suffix).compareTo(right + suffix)
            }
        }
    }
}
